misaligned/misaligned.java

```diff
@@ -1,18 +1,180 @@
+import java.util.LinkedList;
+import java.util.List;
+
+import com.tcq.training.colourCode.ColourCode;
+
 public class Misaligned {
-    static int printColorMap() {
-        String majorColors[] = {"White", "Red", "Black", "Yellow", "Violet"};
-        String minorColors[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-        int i = 0, j = 0;
-        for(i = 0; i < 5; i++) {
-            for(j = 0; j < 5; j++) {
-                System.out.printf("%d | %s | %s\n", i * 5 + j, majorColors[i], minorColors[i]);
-            }
-        }
-        return i * j;
-    }
-    public static void main(String[] args) { 
-        int result = printColorMap();
-        assert(result == 25);
-        System.out.println("All is well (maybe!)");
-    }
+
+	void printColorMap(List<ColourCode> listOfColourCode) {
+		for (ColourCode colourCode : listOfColourCode) {
+			System.out.printf("%d  | %s | %s\n", colourCode.getPairNumber(), colourCode.getMajorColor(),
+					colourCode.getMinorColor());
+		}
+
+	}
+
+	List<ColourCode> generateColourCode() {
+		String majorColors[] = { "White", "Red", "Black", "Yellow", "Violet" };
+		String minorColors[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+		List<ColourCode> listOfColourCode = new LinkedList<ColourCode>();
+		int i = 0, j = 0;
+		for (i = 0; i < 5; i++) {

```

Comment:
The number 5 used in the pair number calculation would lead to confusion or error in future when someone else is maintaining the code.
Also, will this code work if major color or minor color length is other than 5. Please think over it and write test cases accordingly to cover such cases.